homeassistant/components/onkyo/const.py

```diff
@@ -0,0 +1,141 @@
+"""Constants for the Onkyo integration."""
+
+from enum import Enum
+import typing
+from typing import Any, ClassVar, Literal, Self, overload
+
+import pyeiscp
+
+DOMAIN = "onkyo"
+
+DEVICE_INTERVIEW_TIMEOUT = 5
+DEVICE_DISCOVERY_TIMEOUT = 5
+
+CONF_SOURCES = "sources"
+CONF_RECEIVER_MAX_VOLUME = "receiver_max_volume"
+
+type VolumeResolution = Literal[50, 80, 100, 200]
+CONF_VOLUME_RESOLUTION = "volume_resolution"
+CONF_VOLUME_RESOLUTION_DEFAULT: VolumeResolution = 50
+VOLUME_RESOLUTION_ALLOWED: tuple[VolumeResolution, ...] = typing.get_args(
+    VolumeResolution.__value__
+)
+
+OPTION_MAX_VOLUME = "max_volume"
+OPTION_MAX_VOLUME_DEFAULT = 100.0
+
+_INPUT_SOURCE_MEANINGS = {
+    "00": "VIDEO1 ··· VCR/DVR ··· STB/DVR",
+    "01": "VIDEO2 ··· CBL/SAT",
+    "02": "VIDEO3 ··· GAME/TV ··· GAME",
+    "03": "VIDEO4 ··· AUX",
+    "04": "VIDEO5 ··· AUX2 ··· GAME2",
+    "05": "VIDEO6 ··· PC",
+    "06": "VIDEO7",
+    "07": "HIDDEN1 ··· EXTRA1",
+    "08": "HIDDEN2 ··· EXTRA2",
+    "09": "HIDDEN3 ··· EXTRA3",
+    "10": "DVD ··· BD/DVD",
+    "11": "STRM-BOX",
+    "12": "TV",
+    "20": "TAPE ··· TV/TAPE",
+    "21": "TAPE2",
+    "22": "PHONO",
+    "23": "CD ··· TV/CD",
+    "24": "FM",
+    "25": "AM",
+    "26": "TUNER",
+    "27": "MUSIC-SERVER ··· P4S ··· DLNA",
+    "28": "INTERNET-RADIO ··· IRADIO-FAVORITE",
+    "29": "USB ··· USB(FRONT)",
+    "2a": "USB(REAR)",
+    "2b": "NETWORK ··· NET",
+    "2d": "AIRPLAY",
+    "2e": "BLUETOOTH",
+    "2f": "USB-DAC-IN",
+    "30": "MULTI-CH",
+    "31": "XM",
+    "32": "SIRIUS",
+    "33": "DAB",
+    "40": "UNIVERSAL-PORT",
+    "41": "LINE",
+    "42": "LINE2",
+    "44": "OPTICAL",
+    "45": "COAXIAL",
+    "55": "HDMI-5",
+    "56": "HDMI-6",
+    "57": "HDMI-7",
+    "80": "MAIN-SOURCE",
+}
+
+_cmds_base = pyeiscp.commands.COMMANDS
+_cmds = dict(
+    sorted(
+        {
+            **_cmds_base["main"]["SLI"]["values"],
+            **_cmds_base["zone2"]["SLZ"]["values"],
+        }.items()
+    )
+)
+del _cmds["2C"]  # USB(TOGGLE)
+del _cmds["7F"]  # OFF
+
+type InputLibValue = str | tuple[str, ...]
+
+
+class InputSource(Enum):
```

Comment:
This class feels very libraryish and not really part of the integration code

Reply:
I agree, I felt the same already as I was writing it.

But as the library is not that well maintained currently, we cannot really change it.

For the longer term, I plan to rewrite and maintain a library myself. And then we can move all these consts and enums there.

Reply:
I need to double check this with some others

Reply:
I simplified it.